src/pgen/pgen_esm_from_fasta.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import argparse
+from pgen.esm_sampler import ESM_sampler
+from pgen import models
+from pgen.utils import write_sequential_fasta, parse_fasta
+from pathlib import Path
+
+model_map = {"esm1b":models.ESM1b, "esm6":models.ESM6, "esm12":models.ESM12, "esm34":models.ESM34}
+
+def main(input_h, output_p, args):
+    sampler = ESM_sampler(model_map[args.model](),device=args.device)
+    with open(output_p / "specification.tsv","w") as output_h:
+        for line in input_h:
+            line = line.strip().split("\t")
+            if len(line) == 3:
+                print("\t".join(line))
+                print("\t".join(line), file=output_h)
+                name = line[0]
+                line_args = eval(line[1])
+                seeds = parse_fasta(line[2])
+                sequences = sampler.generate(args.num_output_sequences, seed_seq=seeds, batch_size=args.batch_size, **line_args)
```

Comment:
This file is identical to the pgen_esm file with the addition of reading from the fasta